src/wp-includes/rest-api.php

```diff
@@ -479,6 +482,23 @@ function rest_get_server() {
 	/* @var WP_REST_Server $wp_rest_server */
 	global $wp_rest_server;
 
+	/* @var WP_User|null $current_user */
+	global $current_user;
+
+	if ( $current_user instanceof WP_User && 0 === $current_user->ID ) {
```

Comment:
Should this be done in `WP_REST_Server::serve_request`? I don't see why we need to reauthenticate if the REST API isn't actually serving a request.

Reply:
Moved.